File: Java-Core/src/main/java/presto/type/AnyType.java
package presto.type;

import presto.error.SyntaxError;
import presto.runtime.Context;


public class AnyType extends NativeType { 

	static AnyType instance = new AnyType();
	
	public static AnyType instance() {
		return instance;
	}
	
	private AnyType() {
		super("any");
	}

	@Override
	public Class<?> toJavaClass() {
		return Object.class;
	}
	
	@Override
	public IType checkItem(Context context, IType itemType) throws SyntaxError {
		return AnyType.instance(); // needed to support lists in Documents
	}
	
	@Override
	public IType checkMember(Context context, String name) throws SyntaxError {
		return AnyType.instance(); // needed to support members in Documents
	}

	@Override
	public boolean isAssignableTo(Context context, IType other) {
		return (other instanceof AnyType);
	}
	
}

File: Java-CodeServer/src/main/resources/js/prompto/type/AnyType.js
var NativeType = require("./NativeType").NativeType;

function AnyType() {
 	NativeType.call(this, "any");
 	return this;
 }

 AnyType.prototype = Object.create(NativeType.prototype);
 AnyType.prototype.constructor = AnyType;

 AnyType.instance = new AnyType();
	
/*

@Override
public Class<?> toJavaClass() {
	return Object.class;
}

*/

AnyType.prototype.checkItem = function(context, name) {
	return AnyType.instance; // required to support Document items
};

AnyType.prototype.checkMember = function(context, name) {
	return AnyType.instance; // required to support Document members
};

AnyType.prototype.isAssignableTo = function(context, other) {
	return (other instanceof AnyType);
};

exports.AnyType = AnyType;
